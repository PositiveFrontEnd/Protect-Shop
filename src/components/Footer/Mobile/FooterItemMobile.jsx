import React, { useState } from "react";
import ToOpen from '../Svg/open.svg?react'
import PropTypes from 'prop-types'
import cx from 'classnames'


const FooterItem = ({ title, children, className }) => {
    const [isOpenFooterMenu, setOpenFooterMenu] = useState(false)
    const openFooterMenu = () => {
        setOpenFooterMenu(!isOpenFooterMenu)


    }
    return (
        <>
            <div className="footer__title__wrapper">
                <h4 className="footer__title">{title}</h4>
                <button onClick={openFooterMenu} className={`toopen ${isOpenFooterMenu ? 'open-animated' : ''}`}>< ToOpen className={isOpenFooterMenu ? 'flip' : ''} /></button>

            </div>
            {isOpenFooterMenu && (
                <ul className={cx("footer__more-info", className)}>
                    {children}
                </ul>)
            }
        </>
    )
}

FooterItem.defaultProps = {
    isOpenFooterMenu: false
}

FooterItem.propTypes = {
    title: PropTypes.string,
    children: PropTypes.any
}

export default FooterItem