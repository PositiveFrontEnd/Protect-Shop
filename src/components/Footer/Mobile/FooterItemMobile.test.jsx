import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import FooterItemMobile from "./FooterItemMobile";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

describe("Тестування компонента  FooterItemMobile", () => {
    test("Знімок", () => {
        const mockStore = configureStore([thunk]);
        const initialState = {
        };

        const store = mockStore(initialState);
        const data = {
            title: "some title",
            children: "some chuldren",
            className: "some__className"
        }
        const footerItemMobile = render(
            <Provider store={store}>
                <BrowserRouter>
                    <FooterItemMobile data={data} />
                </BrowserRouter>
            </Provider>
        );

        expect(footerItemMobile).toMatchSnapshot();
    });
});
