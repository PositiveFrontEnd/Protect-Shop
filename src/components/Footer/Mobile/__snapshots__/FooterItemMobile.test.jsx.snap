// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестування компонента  FooterItemMobile > Знімок 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="footer__title__wrapper"
      >
        <h4
          class="footer__title"
        />
        <button
          class="toopen "
        >
          <svg
            class=""
            fill="none"
            height="17"
            viewBox="0 0 16 17"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.13828 7.04375L5.10391 6.07812L8.00078 8.975L10.8977 6.07812L11.8633 7.04375L8.00078 10.9062L4.13828 7.04375Z"
              fill="#FEFEFE"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="footer__title__wrapper"
    >
      <h4
        class="footer__title"
      />
      <button
        class="toopen "
      >
        <svg
          class=""
          fill="none"
          height="17"
          viewBox="0 0 16 17"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.13828 7.04375L5.10391 6.07812L8.00078 8.975L10.8977 6.07812L11.8633 7.04375L8.00078 10.9062L4.13828 7.04375Z"
            fill="#FEFEFE"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
